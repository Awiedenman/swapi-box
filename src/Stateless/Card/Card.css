.card {

  height: 300px;
  width: 200px;
  background-color: white;
  margin: 15px 20px;
  text-align: left;
  padding: 20px;
  border: 1px #ffff00 inset;
}

.card selected {
  height: 300px;
  width: 200px;
  background-color: #81d8be;
  margin: 15px 20px;
  text-align: left;
  padding: 20px;
  border: 1px #ffff00 inset;
}

h1 { 
  font-size: 30px;
}

.fav-star:hover {
  color: blue;
}
