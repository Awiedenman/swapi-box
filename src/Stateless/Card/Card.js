import React from 'react';
import './Card.css'
import PropTypes from 'prop-types';

const Card = ( { card, addFavorite, favCardClicked, selectedClass } ) => {
  // debugger;
  // console.log(card)
  const dynamicCards = Object.keys(card.data).map(( value, index) => 
      <p key={index}> 
          { card.data[value] }
          { favCardClicked } </p>
      )

  return (
    <div className={ selectedClass }>
      < div className = "fav-star"
        onClick = { () => addFavorite(card) }> 
        < span role="img" aria-label="favorite button"> ⭐️ </span> 
      </div >
      <h1> { card.name } </h1>
          { dynamicCards }
    </div>
  )

}



Card.propTypes = {
  card: PropTypes.object.isRequired,
  addFavorite: PropTypes.func.isRequired,
  favCardClicked: PropTypes.bool.isRequired,
  selectedClass: PropTypes.string.isRequired
}

export default Card;