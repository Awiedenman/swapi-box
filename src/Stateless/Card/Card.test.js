import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  beforeEach( () => {
    const mockCard ={
      data: {
        Climate: "Climate: frozen",
        Population: "Population: unknown",
        Residents: "Residents: N/A",
        Terrain: "Terrain: tundra, ice caves, mountain ranges",
        name: "Hoth"
      }};

    const mockAddFavoite = jest.fn();
    const mockFavCardClicked = true;
    wrapper = shallow( 
      < Card 
        card={ mockCard }
        addFavorite = { mockAddFavoite }
        favCardClicked = { mockFavCardClicked }
  
      />
    );
  });

  it('should match the snapshot when cards are rendered on the page', () => {
    expect( wrapper ).toMatchSnapshot();
  });
});
