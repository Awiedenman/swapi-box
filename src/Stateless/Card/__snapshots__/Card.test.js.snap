// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot when cards are rendered on the page 1`] = `[Function]`;
